test/r401_test.js

```diff
@@ -0,0 +1,336 @@
+    const claim = pt.findRecords('Claim').find(p => p.getId() === '58cd648a-5f4d-4306-bef4-49ec64c88c63');
+    expect(claim.get('item')[0].get('sequence.value')).to.equal(1);
+  });
+
+  it('should support getting strings', () => {
+    const pt = patientSource.currentPatient();
+    const procedure = pt.findRecords('Procedure').find(p => p.getId() === '03c48dbb-2e69-451d-877a-b3397a9f3d26');
+    expect(procedure.get('status.value')).to.deep.equal('completed');
+  });
+
+  it('should support getting dateTimes', () => {
+    const pt = patientSource.currentPatient();
+    const medReq = pt.findRecords('MedicationRequest').find(p => p.getId() === '622c5788-3028-41fd-a8cb-164f868d4322');
+    const authoredOn = medReq.getDate('authoredOn.value');
+    expect(authoredOn.isDateTime).to.be.true;
+    expect(authoredOn).to.deep.equal(cql.DateTime.parse('2009-12-01T11:18:29-05:00'));
+  });
+
+  it('should support getting dates', () => {
+    const pt = patientSource.currentPatient();
+    const birthDate = pt.get('birthDate.value');
+    expect(birthDate.isDate).to.be.true;
+    expect(birthDate).to.deep.equal(cql.DateTime.parse('2008-11-06').getDate());
+  });
+
+  it('should support getting times', () => {
+    const pt = patientSource.currentPatient();
+    const observation = pt.findRecords('Observation').find(p => p.getId() === '9c15c801-6bb5-47a7-a9db-8bad0cb6aa68');
+    const valueTime = observation.get('value.value');
+    expect(valueTime.isTime()).to.be.true;
+    expect(valueTime).to.deep.equal(cql.DateTime.parse('0000-01-01T18:23:47.376-05:00').getTime());
+  });
+
+  it('should support getting an option of a choice', () => {
+    const pt = patientSource.currentPatient();
+    const condition = pt.findRecords('Condition').find(p => p.getId() === '9934bc4f-58af-4ecf-bb70-b7cc31987fc5');
+    // In STU3, you use the stub of the choice (e.g., onset[x] datetime is retrieved as onset)
+    expect(condition.get('onset.value')).to.deep.equal(cql.DateTime.parse('2009-08-09T12:18:29-04:00'));
+  });
+
+  it('should support getting an option of a choice using explicit name', () => {
+    // This was needed because the ModelInfo used to indicate MedicationRequest's primaryCodePath as medicationCodeableConcept!
+    // Keep it in case anyone tries to load an older ModelInfo that still does that.
+    const pt = patientSource.currentPatient();
+    const medReq = pt.findRecords('MedicationRequest').find(p => p.getId() === '622c5788-3028-41fd-a8cb-164f868d4322');
+    const code = medReq.getCode('medicationCodeableConcept');
+    expect(code).to.deep.equal(new cql.Code('309097', 'http://www.nlm.nih.gov/research/umls/rxnorm', undefined, 'Cefuroxime 250 MG Oral Tablet'));
+  });
+
+  it('should not return wrong type if explicit choice name was requested but data used different type', () => {
+    // This is needed because the ModelInfo indicates MedicationRequest's primaryCodePath as medicationCodeableConcept!
```

Comment:
Similarly, the 4.0.1 modelinfo has `primaryCodePath` `medication`, so I don't think this test is relevant for 4.0.1.